### PR TITLE
Change command line option storage for SamplerFullSolveMultiApp 

### PR DESCRIPTION
**Pull request by jiangwen84**

### CODE REVIEW

**Comment by permcody** [comment]
Looks like you need to use a space between ref and your ticket number. Github completely missed it so we don't have the traceability between your PR and your ticket (either direction). Also, did you intend to close that issue? If so, you should use "closes #<number>" instead. Then we don't have to go back later and figure out that it should have been closed. Thanks!

**Comment by jiangwen84** [comment]
@permcody thanks! I added a space now and I think it links to the issue# now. 

I do not want to close #13879 now since I have some follow-up PRs to address that issue.

File: modules/stochastic_tools/src/controls/MultiAppCommandLineControl.C [hunk 1]
```diff
@@ -106,14 +106,29 @@ MultiAppCommandLineControl::execute()
                  _param_names.size(),
                  ").");
 
-    for (unsigned int row = 0; row < matrix.m(); ++row)
+    if (std::dynamic_pointer_cast<SamplerFullSolveMultiApp>(_multi_app) == nullptr)
```
Comment (aeslaughter) [inline modules/stochastic_tools/src/controls/MultiAppCommandLineControl.C]: Can you explain what you are trying to accomplish?

Comment (jiangwen84) [reply modules/stochastic_tools/src/controls/MultiAppCommandLineControl.C]: The command line option is like name_a = 1, name_b = 2, name_c = 3..... The name is the same for each sampler and only the values are different. If we store them all as strings, the mem usage is huge. So I think I just need to store only names into the cli_arg and them in the ``getCommandLineArgsParamHelper`` I added the name to values together. ``getCommandLineArgsParamHelper`` is a virtual function defined in ``MultiApp`, so I do no want to change this for standard multiApp but just for ``SamplerFullSolveMultiApp``.

Comment (aeslaughter) [reply modules/stochastic_tools/src/controls/MultiAppCommandLineControl.C]: OK, can you write some detailed comments above this if statement and in the SamplerFullSolveMultiApp::getCommandLineArgsParamHelper so we (I) can remember why this is done.

Comment (jiangwen84) [reply modules/stochastic_tools/src/controls/MultiAppCommandLineControl.C]: Sure, I will do it. 

Do you have any other comments that I need to address?

Comment (aeslaughter) [reply modules/stochastic_tools/src/controls/MultiAppCommandLineControl.C]: No, I am good with what you did.

**Comment by jiangwen84** [comment]
comments were addressed.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**